### PR TITLE
Simple perflevel detection + separate algo_id

### PR DESCRIPTION
**Pull request by LuKePicci**
nVidia editable OC performance level detection rewritten for readability

### CODE REVIEW

File: 3main [hunk 1]
```diff
@@ -155,31 +158,18 @@ then
...
 
-if grep -q "1050" /tmp/tempa; then
-  ___1050_or_1050ti="YES"
-fi
-
-if grep -q 'P106' /tmp/tempa; then
-  ___1050_or_1050ti="YES"
-  P106_100="YES"
+if grep -q -E 'P104|P102' /tmp/tempa; then
+  GPU_OC_PROFILE="1 ${GPU_OC_PROFILE}"
```
Comment (papampi) [inline 3main:164]: Nice idea.

**Comment by papampi** [comment]
Will give it some tests ...

**Review by papampi — DISMISSED** [review]

**Review by papampi — DISMISSED** [review]

File: 0algo_id [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  ALGO=LBRY
+elif [[ $COIN == DMD ]]
+then
+  ALGO=DMD-GR
+elif [[ $COIN == GRS || $COIN == MPH_GRS ]]
+then
+  ALGO=GRS
+elif [[ $COIN == XMY ]]
+then
+  ALGO=MYR-GR
```
Comment (papampi) [inline 0algo_id]: As @sizzlephizzle reminded, hyphen will break the variable, MYR-GR and DMD-GR should change to underscore

**Review by papampi — DISMISSED** [review]

**Comment by papampi** [comment]
Does 0miner need source 0algo_id too?

**Comment by LuKePicci** [comment]
Nope, 0miner is only launched from 3main, and it already source 0algo_id

**Review by papampi — DISMISSED** [review]

File: 3main [hunk 3]
```diff
@@ -63,8 +63,11 @@
 #          AUTO_SWITCH generalized implementetion
 # v=0021 : papampi
 #          Algo identification edits for Algo mining unification
+# v=0022 : LuKePicci
+#          Reworked OC PerfLevel detection, 0algo_id
 
 source ${NVOC}/1bash
+
 nvOC_ver="nvOC v0019-2.1 - Community Release"   # Do not edit this
 nvOC_3main_ver="v0019-2.1.0021"                  # Do not edit this
```
Comment (LuKePicci) [inline 3main]: I forgot to increment the revision, than should be fine. Please do it for me, I cannot right now.

**Review by papampi — DISMISSED** [review]

**Review by papampi — DISMISSED** [review]

**Comment by papampi** [comment]
@LuKePicci 
You want make more changes or tests?

**Review by papampi — APPROVED** [review]

**Pull request merged**